### ISSUE TITLE
[Feature request] add notifyDayChanged(long day)

### ISSUE THREAD

**Issue by dgudim** [kizitonwose/Calendar#442]
Currently, you can only refresh a `LocalDate` or `CalendarDay` via `notifyDateChanged` and `notifyDayChanged`. But in my case the event indicator is visible on `InDate`/`OutDate` and if the user updates the event, corresponding `InDate`/`OutDate` of the next/previous month needs to be rebound to reflect changes. Currently i use something like this, but it's clearly sub-optimal.

```java
    public void notifyDayChanged(long targetDayUTC) {
        if (targetDayUTC < getFirstLoadedDayUTC() || targetDayUTC > getLastLoadedDayUTC()) {
            // day is not loaded, skip it
            return;
        }
        rootCalendarView.notifyDateChanged(date, DayPosition.InDate);
        rootCalendarView.notifyDateChanged(date, DayPosition.OutDate);
        rootCalendarView.notifyDateChanged(date, DayPosition.MonthDate);
    }
```

I want a method that will do all of that but efficiently

**Comment by kizitonwose** [comment]
That sounds like a good idea, I'll see if it can go into the next release. The difficult thing I see here is figuring out the appropriate name for this function.

**Comment by dgudim** [comment]
Thanks. Yeah, naming functions is often hard. I have a few suggestions though
1. Add it as an optional argument to existing function(s)
2. Add a value to `DayPosition` enum, something like `FullDate`
3. `notifyDateChangedFull`
4. `notifyDateChangedComplete`

And I have one question: How can I determine which days are loaded and will not be rebound on scroll and which ones will? So basically first and last loaded day. Currently I subtract 60 from the first visible day and add 60 to the last, but I couldn't find a way to know for sure.

**Comment by kizitonwose** [comment]
> How can I determine which days are loaded and will not be rebound on scroll and which ones wil

Since `findFirstVisibleMonth` and `findLastVisibleMonth` use the LayoutManger functions `findFirstVisibleItemPosition` and `findLastVisibleItemPosition` respectively, you can use these to determine the range. However, according to [here](https://developer.android.com/reference/androidx/recyclerview/widget/LinearLayoutManager#findFirstVisibleItemPosition()), it is possible that there are cached positions that are omitted in this call so it is not 100% reliable.

**Comment by dgudim** [comment]
That's a pity. I'll keep using my method then

**Comment by kizitonwose** [comment]
In the latest snapshot, you can call any of these to reload all available day positions.

```kotlin
val date = LocalDate.now() // your desired date
calendarView.notifyDateChanged(date, *DayPosition.values())
// OR
calendarView.notifyDateChanged(date, DayPosition.InDate, DayPosition.MonthDate, DayPosition.OutDate)
```

This also means you can do any kind of partial reload - for example, you can reload only the in-date and out-date.

```kotlin
calendarView.notifyDateChanged(date, DayPosition.InDate, DayPosition.OutDate)
```

**Comment by dgudim** [comment]
Nice, that's even better that my suggestion, great.

**Comment by kizitonwose** [comment]
Released in [2.2.0](https://github.com/kizitonwose/Calendar/releases/tag/2.2.0)
